com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/UploadAppStepTest.java

```diff
@@ -244,11 +245,11 @@ public File saveToFile(String fileName, InputStreamProducer streamProducer) thro
             }
 
             @Override
-            InputStreamProducer getInputStreamProducer(InputStream appArchiveStream, String fileName) throws SLException {
+            InputStreamProducer getInputStreamProducer(InputStream appArchiveStream, String fileName, long maxStreamSize) throws SLException {
                 if (!fileName.equals(APP_FILE)) {
-                    return super.getInputStreamProducer(appArchiveStream, fileName);
+                    return super.getInputStreamProducer(appArchiveStream, fileName, maxStreamSize);
                 }
-                return new InputStreamProducer(getClass().getResourceAsStream(APP_FILE), fileName) {
+                return new InputStreamProducer(getClass().getResourceAsStream(APP_FILE), fileName, Configuration.getInstance().getMaxResourceFileSize()) {
                     @Override
```

Comment:
Configuration.getInstance().getMaxResourceFileSize() is used everywhere think of a way to avoid this

Reply:
No reason to avoid. This is Singleton class so this is the way to retrieve data.

Reply:
Configuration is injected in UploadAppStep, no need to use getInstance() here. You can access it dynamically. It is easier to mock that way.